### PR TITLE
Fix new 3.4 control() to work properly.

### PR DESCRIPTION
**Pull request by dereuromark**
Fixes https://github.com/FriendsOfCake/bootstrap-ui/pull/173

### CODE REVIEW

**Comment by dereuromark** [comment]
Well, it's not quite the proper BC fix here :) 3.4 works now, but 3.3 and before don't.

File: src/View/Helper/FormHelper.php [hunk 1]
```diff
@@ -169,8 +169,26 @@ public function submit($caption = null, array $options = [])
      * @param string $fieldName This should be "Modelname.fieldname".
      * @param array $options Each type of input takes different options.
      * @return string Completed form widget.
+     * @deprecated Use control() instead.
      */
-    public function input($fieldName, array $options = [])
+    public function input($fieldName, array $options = []) {
+        return $this->control($fieldName, $options);
```
Comment (josegonzalez) [inline src/View/Helper/FormHelper.php]: PHPCS violation here.

File: src/View/Helper/FormHelper.php [hunk 2]
```diff
@@ -246,7 +264,7 @@ public function input($fieldName, array $options = [])
             );
         }
 
-        $result = parent::input($fieldName, $options);
+        $result = parent::control($fieldName, $options);
```
Comment (josegonzalez) [inline src/View/Helper/FormHelper.php]: Could do a `method_exists` check to add support for 3.3 and below.

**Comment by ADmad** [comment]
Tests failing.

Adding the new `control()` method and maintaining BC without code duplication is gonna be tricky.

**Comment by dereuromark** [comment]
Good idea, @josegonzalez

**Comment by ADmad** [comment]
Nice, tests passing now. I'll try it in one of my apps to check for BC and then merge.

File: src/View/Helper/FormHelper.php [hunk 3]
```diff
@@ -102,6 +109,11 @@ public function __construct(View $View, array $config = [])
 
         $this->_defaultWidgets = $this->_widgets + $this->_defaultWidgets;
 
+        $this->_controlMethod = 'input';
```
Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: You could also default the property to "input" - and only overwrite here if necessary.
That would get rid of this line.

Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: But shouldn't it be the other way around: control() now by default, and if !method exists we fall back to the old input?

Comment (josegonzalez) [inline src/View/Helper/FormHelper.php]: I didn't want to add the extra negation. Ever so slightly faster :P

Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: For me it's more about the logical "thats the status quo" and "thats the old bc fallback" kind of way. the time speed is irrelevant anyway, as you well know :)

Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: But no big deal either way.

**Comment by ADmad** [comment]
Since the public API has changed it's prudent to bump minor version in new release right?  I just released 0.6.0 yesterday :stuck_out_tongue:

**Comment by dereuromark** [comment]
It was just a method being added. I think a bugfix version would also be reasonable.
Either way, you are still below 1.0 - a new minor also doesn't hurt :)

**Comment by dereuromark** [comment]
My question still stands: How can we test Cake plugins beforehand towards the 3.next in Travis?
That would solve most of those things when the next minor comes out.

**Comment by ADmad** [comment]
> How can we test Cake plugins beforehand towards the 3.next in Travis?

Dynamically switch the core version in `.travi.yml` similar to how it's done [here](https://github.com/cakephp/cakephp/blob/3.next/.travis.yml#L91) for the codesniffer version.

**Comment by dereuromark** [comment]
Shouldn't we promote that for all plugins for Cake, FriendsOfCake and the rest of the ecosystem?

**Comment by ADmad** [comment]
If someone wants to put in the effort sure.

**Pull request merged**